boards/st/nucleo_u031r8/doc/index.rst

```diff
@@ -0,0 +1,268 @@
+  - PLL for system clock, ADC
+
+- Security:
+
+  - Customer code protection
+  - Robust read out protection (RDP): 3 protection level states
+    and password-based regression (128-bit PSWD)
+  - Hardware protection feature (HDP)
+  - Secure boot
+  - True random number generation, candidate for NIST SP 800-90B certification
+  - Candidate for Arm |reg| PSA level 1 and SESIP level 3 certifications
+  - 5 passive anti-tamper pins
+  - 96-bit unique ID
+
+- Up to 9 timers, RTC, and 2 watchdogs :
+
+  - 1x 16-bit advanced motor-control, 1x 32-bit and 3x 16-bit general purpose,
+    2x 16-bit basic, 2x low-power 16-bit timers (available in Stop mode),
+    2x watchdogs, SysTick timer
+  - RTC with hardware calendar, alarms and calibration
+
+- Up to 16 communication peripherals:
+
+  - 6x USARTs/LPUARTs (SPI, ISO 7816, LIN, IrDA, modem)
+  - 3x I2C interfaces supporting Fast-mode and Fast-mode Plus (up to 1 Mbit/s)
+  - 2x SPIs, plus 4x USARTs in SPI mode
+  - IRTIM (Infrared interface)
+
+- Rich analog peripherals (independent supply):
+
+  - 1x 12-bit ADC (0.4 µs conversion time), up to 16-bit with hardware oversampling
+  - 1x 12-bit DAC output channel, low-power sample and hold
+  - 1x general-purpose operational amplifier with built-in PGA (variable gain up to 16)
+  - 1x ultra-low-power comparator
+
+- ECOPACK2 compliant packages
+
+More information about STM32U031R8 can be found here:
+
+- `STM32U031R8 on www.st.com`_
+- `STM32U031R8 reference manual`_
+
+Supported Features
+==================
+
+The Zephyr _nucleo_u031r8_ board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/st/nucleo_u031r8/nucleo_u031r8_defconfig`
+
+
+Connections and IOs
+===================
+
+Nucleo U031R8 Board has 10 GPIO controllers. These controllers are responsible
+for pin muxing, input/output, pull-up, etc.
+
+For more details please refer to `STM32U031 User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- DAC1_OUT1 : PA4
+- I2C1_SCL : PB8
+- I2C1_SDA : PB9
+- LPUART_1_TX : PA2
+- LPUART_1_RX : PA3
+- SPI1_NSS : PA4
+- SPI1_SCK : PA5
+- SPI1_MISO : PA6
+- SPI1_MOSI : PA7
+- UART_2_TX : PA2
+- UART_2_RX : PA3
+- USER_PB : PC13
```

Comment:
Please revise this list, at the moment only one led, one button and two UARTs are present in dts